pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java

```diff
@@ -28,10 +28,31 @@
   HEALTHCHECK_BAD_CALLS("healthcheck", true),
   HEALTHCHECK_OK_CALLS("healthcheck", true),
   QUERIES("queries", false),
+
+  // These metrics track the exceptions caught during query execution.
+  // PQL compile phase.
   REQUEST_COMPILATION_EXCEPTIONS("exceptions", true),
-  REQUEST_FETCH_EXCEPTIONS("exceptions", false),
-  REQUEST_DESERIALIZATION_EXCEPTIONS("exceptions", false),
+  //Query validation phase.
+  QUERY_VALIDATION_EXCEPTIONS("exceptions", false),
+  // Scatter phase.
+  RESOURCE_MISSING_EXCEPTIONS("exceptions", false),
+  NO_SERVER_FOUND_EXCEPTIONS("exceptions", false),
+  // Gather phase.
+  RESPONSE_FETCH_EXCEPTIONS("exceptions", false),
+  // Response deserialize phase.
+  DATA_TABLE_DESERIALIZATION_EXCEPTIONS("exceptions", false),
+  // Reduce responses phase.
+  RESPONSE_MERGE_EXCEPTIONS("exceptions", false),
+
+  // These metrics track the number of partial results.
+  PROCESSING_EXCEPTION_CAUGHT("exceptions", false),
```

Comment:
pls document what these metrics mean. From name, I can not figure what is exactly tracked by processing_exception_caught.

Why "caught" part in the name ?